docs/advanced-topics/reconnecting-mid-game.md

```diff
@@ -0,0 +1,28 @@
+---
+id: reconnecting-mid-game
+title: Reconnecting Mid-Game
+---
+
+In a multiplayer game, it is possible that a client gets disconnected from the server, for example if they have network issues, or if their application or device crashes. When it happens, you might want to give the ability to your players to reconnect to the game.
+
+To allow clients to reconnect in the middle of an active game, there are some considerations to have in mind.
+- If clients have a specific state and some in-game data associated with them and you want them to reconnect with that same state and that same data, you should implement a mechanism to keep that state in memory and associate it to that client when they reconnect (see [Session Management](./session-management.md))
+- If using scene management, keep in mind that connecting to a game launches the synchronization process. If you're using additive scenes, this means that if the client's active main scene is the same as the server's, it will load all additive scenes that are currently loaded on the server, without unloading anything on the client. If that is a behavior you do not want, you can unload those scenes on the client when they disconnect, or before attempting to reconnect, or you can use the `NetworkSceneManager.VerifySceneBeforeLoading` callback to prevent loading scenes that are already loaded on the client.
+- If your game requires it, you should make sure that the client's state is properly reset and ready to connect before attempting to reconnect. This could require resetting some external services, for example.
+
+
+### Reconnecting automatically
+
+To make the experience smoother for the players, you can have clients automatically attempt to reconnect to a game when losing connection unexpectedly. To implement this, you'll have to do the following:
```

Comment:
Automatically reconnecting a client that has lost its connection unexpectedly improves the player experience.
Guidelines for reconnecting a client automatically:
- Determine what will initiate the reconnection process.
  - For example, it could be the `NetworkManager.OnClientDisconnectCallback` callback or it could be some other unique event specific to your product.
- Wait for `NetworkManager` to properly shut down before attempting to reconnect. You can use the `NetworkManager.ShutdownInProgress` property for this.
  - Typically it takes at least 1 frame for this to happen
- Add additional checks to assure  your automatic client reconnection solution it not triggered under the wrong conditions.
  - For example, if a client quits a game purposefully, if the server shuts down normally as the game session ends, if a client gets kicked from a game or if they get denied during connection approval you do not want the client to automatically reconnect.